File: HyungJoonLEE/2022-07-29-Rotate-Image.c
#include "common.h"

void rotate(int** matrix, int matrixSize, int* matrixColSize){
    for(int i = 0; i < matrixSize / 2; i++){
        for(int j = i; j < matrixSize - i - 1; j++){
            int a = matrix[i][j];
            int b = matrix[j][matrixSize - i - 1];
            int c = matrix[matrixSize - i - 1][matrixSize -j -1];
            int d = matrix[matrixSize - j - 1][i];
            matrix[i][j] = d;
            matrix[j][matrixSize - i -1] = a;
            matrix[matrixSize - 1 - i][matrixSize - 1 - j] = b;
            matrix[matrixSize - 1 - j][i] = c;
        }
    }
}

File: somijeon/2022-07-29-rotate-image.js
// You are given an n x n 2D matrix representing an image,
//todo: rotate the image by 90 degrees (clockwise).

// You have to rotate the image in-place, which means you have to modify the input 2D matrix directly. DO NOT allocate another 2D matrix and do the rotation.

var rotate = function (matrix) {
  let col = 1;
  for (let i = 0; i < matrix.length; i++) {
    for (let j = col++; j < matrix[0].length; j++) {
      [matrix[i][j], matrix[j][i]] = [matrix[j][i], matrix[i][j]];
    }
  }
  for (let r of matrix) {
    r.reverse();
  }
};
